### ISSUE TITLE
Strictness option to use precise returns types for certain stdlib functions

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#6214]
The return type of `int.__pow__` is `Any`, since the precise type depends on the value of the right operand. Maybe we should have a strictness option that makes the return type `float` instead. This probably shouldn't be enabled by default, since it can generate false positives. We could use the same thing for `pow()` and perhaps some other commonly used stdlib functions -- and maybe even for JSON return values from `json` once we have recursive types.

See #6210 for context.

**Comment by ilevkivskyi** [comment]
The question is were the alternative (more strict) types would be stored? If also somewhere in typeshed, then we might need to have some syntax for this (or use the `Annotated[...]` proposal, which is kind of almost universal syntax):
```python
from typing import Annotated
from mypy_extensions import MypyStrict

class int:
    def __pow__(self, other: float) -> Annotated[Any, MypyStrict[float]]: ...
```

**Comment by JukkaL** [comment]
I'm not sure what's the best way to implement the strict types. If only a handful of functions are affected, we could use the plugin system. If there are a large number of them, the `Annoted[...]` seems worth considering.

**Comment by ilevkivskyi** [comment]
Make we should collect some preliminary list of possible candidates? @JelleZijlstra @srittau do you have any candidates in typeshed where a return type is `Any` to avoid false positives?

**Comment by JelleZijlstra** [comment]
python/typing#1096 is a more recent iteration of this idea.
